### PR TITLE
Fix Tooltip Rendering Issue by Upgrading react-charts

### PR DESCRIPTION
**Pull request by TimothyAsirJeyasing**
This PR addresses the tooltip rendering issues
encountered in react-charts@7.1.0.
The issue has been resolved in react-charts@7.3.0.

https://bugzilla.redhat.com/show_bug.cgi?id=2281587

### CODE REVIEW

**Comment by GowthamShanmugam** [comment]
Please check the build failure

**Comment by SanjalKatiyar** [comment]
> update `shared` dependencies as well...

this is still missing...

**Comment by SanjalKatiyar** [comment]
why do we need changes in analyse file ??

**Comment by TimothyAsirJeyasing** [comment]
> why do we need changes in analyse file


We updated the analyzeTest.ts file to fix two key issues: First, after upgrading react-charts, the script needed adjustments to accurately analyze the new build artifacts and avoid misinterpretations during the build process. Second, it was encountering a memory limit error ("Cannot create a string longer than 0x1fffffe8 characters") when processing large files, which we've optimized.

**Comment by SanjalKatiyar** [comment]
> We updated the analyzeTest.ts file to fix two key issues: First, after upgrading react-charts, the script needed adjustments to accurately analyze the new build artifacts and avoid misinterpretations during the build process. Second, it was encountering a memory limit error ("Cannot create a string longer than 0x1fffffe8 characters") when processing large files, which we've optimized.

You have not "optimized" anything, rather just increases the base size to make sure there are no errors while running the script. I don't think all the changes in the analyse file is necessary, plz check again.
https://github.com/red-hat-storage/odf-console/pull/1520 should do the required optimisation.

**Comment by SanjalKatiyar** [comment]
fix ur commits in this PR... also check why tests are failing...

File: package.json [hunk 1]
```diff
@@ -89,6 +89,7 @@
     "js-base64": "^2.1.9",
     "js-yaml": "^3.13.1",
     "lodash-es": "^4.17.21",
+    "mocha": "^10.7.3",
```
Comment (alfonsomthd) [inline package.json:92]: @TimothyAsirJeyasing Is this dependency required?

**Comment by alfonsomthd** [comment]
@TimothyAsirJeyasing After some investigation about the E2E errors I found that the chart was crashing (at least when you create a storage system and there is no total/available/used capacity yet).
The cause was the tooltip that depends on `victory-tooltip` dep.
Adding this under the `resolutions` section of `package.json` solved the issue in my environment:
`"@patternfly/react-charts/victory-tooltip": "^37.1.1"`
Probably you'll need to also add this in the shared library.
Please test that all charts affected by this PR work as expected.
@SanjalKatiyar FYI.

**Comment by SanjalKatiyar** [comment]
If version upgrade is causing multiple issues, can't we try to fix tooltip based on some CSS property ?? (like z-index or something similar)

**Comment by TimothyAsirJeyasing** [comment]
The react-charts version updated in https://github.com/red-hat-storage/odf-console/pull/1650

**Comment by alfonsomthd** [comment]
> The react-charts version updated in #1650

@SanjalKatiyar this PR can be closed once @TimothyAsirJeyasing confirms that the issue is fixed in master.
The bug should point at https://github.com/red-hat-storage/odf-console/pull/1650 as its fix.

**Pull request closed without merging**